service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/AddGatekeepingOrderController.java

```diff
@@ -0,0 +1,154 @@
+import uk.gov.hmcts.reform.fpl.service.docmosis.GatekeepingOrderGenerationService;
+import uk.gov.hmcts.reform.fpl.service.sdo.GatekeepingOrderEventNotificationDecider;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.CASE_TYPE;
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.JURISDICTION;
+import static uk.gov.hmcts.reform.fpl.enums.DocmosisTemplates.SDO;
+import static uk.gov.hmcts.reform.fpl.enums.OrderStatus.SEALED;
+import static uk.gov.hmcts.reform.fpl.enums.State.CASE_MANAGEMENT;
+import static uk.gov.hmcts.reform.fpl.model.common.DocumentReference.buildFromDocument;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDetailsHelper.removeTemporaryFields;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.DATE;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.formatLocalDateToString;
+
+@Api
+@RestController
+@RequestMapping("/callback/add-gatekeeping-order")
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class AddGatekeepingOrderController extends CallbackController {
+    private final DocumentService documentService;
+    private final GatekeepingOrderGenerationService gatekeepingOrderGenerationService;
+    private final CoreCaseDataService coreCaseDataService;
+    private final GatekeepingOrderEventNotificationDecider notificationDecider;
+    private final NoticeOfProceedingsService nopService;
+    private final GatekeepingOrderService service;
+
+    @PostMapping("/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        Map<String, Object> data = caseDetails.getData();
+        CaseData caseData = getCaseData(caseDetails);
+
+        if (caseData.getAllocatedJudge() != null) {
+            data.put("gatekeepingOrderIssuingJudge", service.setAllocatedJudgeLabel(caseData.getAllocatedJudge(),
+                caseData.getGatekeepingOrderEventData().getGatekeepingOrderIssuingJudge()));
+        }
+
+        return respond(caseDetails);
+    }
+
+    @PostMapping("/generate-draft/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleGenerateDraftMidEvent(
+        @RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        DocmosisStandardDirectionOrder templateData = gatekeepingOrderGenerationService.getTemplateData(caseData);
+        Document document = documentService.getDocumentFromDocmosisOrderTemplate(templateData, SDO);
+
+        caseDetails.getData().put("saveOrSendGatekeepingOrder", service.buildSaveOrSendPage(caseData, document));
+
+        return respond(caseDetails);
+    }
+
+    //Need to redesign for upload + urgent upload routes
```

Comment:
some handler that takes in the route from page 1 and determines what to do based on that

Reply:
sort out when we port over existing functions